ecommerce/pricing/lib/pricing/services.rb

```diff
@@ -45,7 +45,31 @@ def initialize(event_store)
 
     def call(cmd)
       @repository.with_aggregate(Offer, cmd.aggregate_id) do |order|
-        order.change_discount(Discounts::PercentageDiscount.new(cmd.amount))
+        order.change_discount(Discounts::GENERAL_DISCOUNT, Discounts::PercentageDiscount.new(cmd.amount))
```

Comment:
Just to verbalise my thoughts: I wonder if the information about the type of the discount is needed here. I don't know for now, but I think I would transfer this information through the `PercentageDiscount` class. That way we don't have to change method's interface. And it would be simple to add some default (such as `GENERAL_DISCOUNT`). What do you think @marlena-b?

Reply:
That was my first thought as well. The main problem is that once we add it to `PercentageDiscount` class it no longer is a value object. There can be only one discount of a specific type assigned to the order so type is actually an identifier of a discount.

This also leads to problems like what type should we use if we want to add two discounts? Or what type should be used for the initial accumulator in the inject here: `discounts.values.inject(Discounts::NoPercentageDiscount.new, :add).apply(catalog_price_for_single)`? These problems made me feel it is not a right design to add it to this class.

Reply:
> The main problem is that once we add it to `PercentageDiscount` class it no longer is a value object. There can be only one discount of a specific type assigned to the order so type is actually an identifier of a discount.

The type would still be a type. If only one discount of specifiic type can be assigned to an order this is just a rule that can be exectued within the order class. But I wouldn't say that extending `PercentageDiscount` by type would make it an entity.

I assume that would still be true:
```ruby
PercentageDiscount.new(type: GENERAL, amount: 100).eql? PercentageDiscount.new(type: GENERAL, amount: 100)
```

I'll refer to the second part of your reply later :)

Reply:
Ok, I can give it another shot 👍  I will merge this PR and fix it in another one.